src/trust.js

```diff
@@ -92,61 +92,79 @@ export default function createTrustModule(web3, contracts, utils) {
-                  },
+                const newConnection = {
+                  safeAddress: trusterAddress,
+                  isIncoming: false, // default
+                  isOutgoing: true,
+                  mutualConnections: [], // default
                 };
-              },
-              {},
-            );
-
-            // Select mutual connections between our related safes and safes they trust
-            connections.forEach(
-              ({ canSendTo, canSendToAddress, user, userAddress }) => {
-                const safe = canSendTo || user;
-                const safeAddress = canSendToAddress || userAddress;
-                const checksumSafeAddress =
-                  web3.utils.toChecksumAddress(safeAddress);
-
-                // Calculate mutual connections if they do not exist yet
-                if (safe && !network[checksumSafeAddress].mutualConnections) {
-                  network[checksumSafeAddress].mutualConnections =
-                    safe.incoming.reduce((acc, curr) => {
-                      const target = web3.utils.toChecksumAddress(
-                        curr.userAddress,
-                      );
-
-                      // If it is a mutual connection and is not self
-                      return network[target] && curr.userAddress !== safeAddress
-                        ? [...acc, target]
-                        : acc;
-                    }, []);
+                network[trusterAddress] = newConnection;
+              });
+            // 2 Add direct trustee connections
+            trustees &&
+              trustees.forEach((trustee) => {
+                // incoming capacity, outgoing trust
+                const trusteeAddress = web3.utils.toChecksumAddress(
+                  trustee.userAddress,
+                );
+                const existingConnection = network[trusteeAddress];
+                if (existingConnection) {
+                  // trusterOfTrustee is already trusting "me" - update record
+                  existingConnection.isIncoming = true;
+                } else {
+                  // new record
+                  const newConnection = {
+                    safeAddress: trusteeAddress,
+                    isIncoming: true,
+                    isOutgoing: false, // default
+                    mutualConnections: [], // default
+                  };
+                  network[trusteeAddress] = newConnection;
                 }
-              },
-            );
-
-            result = Object.values(network);
+                // 3 Add mutual connections
+                const trustersOfTrustee = trustee.user.outgoing;
+                trustersOfTrustee &&
+                  trustersOfTrustee.forEach((trusterOfTrustee) => {
+                    const ttAddress = web3.utils.toChecksumAddress(
+                      trusterOfTrustee.canSendToAddress,
+                    );
+                    if (ttAddress !== trusteeAddress) {
```

Comment:
Are the incoming and outgoing datasets from the subgraph include connections to itself?  If so, this `if` is not necessary.

Reply:
Yes it was necessary, i did it first without this check